src/lib/ngsi/ContextElement.cpp

```diff
@@ -111,6 +111,32 @@ std::string ContextElement::render(ConnectionInfo* ciP, RequestType requestType,
 
 /* ****************************************************************************
 *
+* ContextElement::toJson - 
+*/
+std::string ContextElement::toJson(ConnectionInfo* ciP, NotificationFormat notifyFormat)
```

Comment:
Well, these functions, toJson() and render() have (with the exception of 'symmetry') the input they need to make their job.   Another way of doing this is to create N function without parameters and have the job done in N places (without ifs) instead.   This give faster and more readable code and I am almost always in favor of that.

Much less maintainable though ...

You are the one that puts maintainability on top of your list, not me ... ;-)